document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -336,8 +365,10 @@ public static String parseNonCompositeFilter(
         throw new UnsupportedOperationException(UNSUPPORTED_QUERY_OPERATION);
     }
 
-    filterString.append(sqlOperator);
-    if (value != null) {
+    if (!isInOP) {
```

Comment:
Can we check if we can simplify these conditions (or at least create different blocks)?